src/components/TaskItem/Drawer/Drawer.tsx

```diff
@@ -40,104 +42,151 @@ export function TaskItemDrawer({
 }: TaskItemDrawerProps): ReactElement {
   const { enqueueSnackbar } = useSnackbar()
   const { mutateAsync } = useUpdateProcess()
+  const [taskStatus, setTaskStatus] = useState(task.status)
+  const [postType, setPostType] = useState<string>()
 
-  function onClick(status: Task['status']): () => Promise<void> {
-    return async function (): Promise<void> {
-      const taskLists = process.taskLists
-      const taskIndex = findIndex(taskList.tasks, ['_id', task._id])
-      taskList.tasks[taskIndex] = { ...task, status }
-      const taskListIndex = findIndex(taskLists, ['title', taskList.title])
-      taskLists[taskListIndex] = taskList
+  async function handleSubmit(): Promise<void> {
+    const taskLists = process.taskLists
+    const taskIndex = findIndex(taskList.tasks, ['_id', task._id])
+    taskList.tasks[taskIndex] = {
+      ...task,
+      status: taskStatus
+    }
+    const taskListIndex = findIndex(taskLists, ['title', taskList.title])
+    taskLists[taskListIndex] = taskList
+
+    try {
+      await mutateAsync({
+        _id: process._id,
+        definition: process.definition,
+        taskLists
+      })
+      enqueueSnackbar('Faith step updated successfully!', {
+        variant: 'success'
+      })
+      handleDialogClose()
+    } catch {
+      enqueueSnackbar('Failed to update faith step. Please try again!', {
+        variant: 'error'
+      })
+    }
+  }
+
+  function handleDialogClose(): void {
+    setPostType(undefined)
+  }
 
-      try {
-        await mutateAsync({
-          _id: process._id,
-          definition: process.definition,
-          taskLists
-        })
-        enqueueSnackbar('Faith step updated successfully!', {
-          variant: 'success'
-        })
-      } catch {
-        enqueueSnackbar('Failed to update faith step. Please try again!', {
-          variant: 'error'
-        })
+  function handleClick(status: Task['status']): () => void {
+    return function (): void {
+      setTaskStatus(status)
+      const postType = getPostType(status)
+      setPostType(postType)
+      if (postType == null) {
+        handleSubmit()
```

Comment:
just to verify this worked right? Hoping that setTaskStatus here updates status before this method but I don't think so.

```suggestion
        handleSubmit(status)
```